src/plugins/js-eval/jsEvalPlugin.js

```diff
@@ -1,54 +1,41 @@
+
+const helpMsg = `n> node stable, h> node --harmony, b> babel, s> node vm.Script, m> node vm.SourceTextModule, e> engine262`;
 
 // default jseval run command
 const CMD = ['node', '--no-warnings', '/run/run.js'];
 const CMD_SHIMS = ['node', '-r', '/run/node_modules/airbnb-js-shims/target/es2019', '--no-warnings', '/run/run.js'];
 const CMD_HARMONY = ['node', '--harmony', '--experimental-vm-modules', '--experimental-modules', '--no-warnings', '/run/run.js'];
 
+
 const jsEvalPlugin = async ({ mentionUser, respond, message, selfConfig = {} }) => {
   if (!/^[nhbsme?]>/.test(message)) return;
   const mode = message[0];
   if (mode === '?') return respond((mentionUser ? `${mentionUser}, ` : '') + helpMsg);
   let code = message.slice(2);
 
-  if (mode === 'b') {
-    code = (await babel.transformAsync(code, {
-      parserOpts: {
-        allowAwaitOutsideFunction: true,
-        allowReturnOutsideFunction: true,
-      },
-      plugins: [
-        '@babel/plugin-transform-typescript',
-        '@babel/plugin-transform-modules-commonjs', // required by dynamicImport
-        ['@babel/plugin-proposal-decorators', { decoratorsBeforeExport: false }], // must be before class-properties https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babel-plugin-proposal-class-properties
-        '@babel/plugin-proposal-class-properties',
-        '@babel/plugin-proposal-do-expressions',
-        '@babel/plugin-proposal-export-default-from',
-        '@babel/plugin-proposal-export-namespace-from',
-        '@babel/plugin-proposal-function-sent',
-        '@babel/plugin-proposal-function-bind',
-        '@babel/plugin-proposal-json-strings',
-        '@babel/plugin-proposal-logical-assignment-operators',
-        '@babel/plugin-proposal-nullish-coalescing-operator',
-        '@babel/plugin-proposal-numeric-separator',
-        '@babel/plugin-proposal-optional-catch-binding',
-        '@babel/plugin-proposal-optional-chaining',
-        '@babel/plugin-proposal-partial-application',
-        ['@babel/plugin-proposal-pipeline-operator', { proposal: 'minimal' }],
-        '@babel/plugin-proposal-throw-expressions',
-        '@babel/plugin-proposal-dynamic-import',
-        '@babel/plugin-syntax-bigint',
-        '@babel/plugin-syntax-import-meta',
-      ]
-    })).code;
+  const hasMaybeTLA = /\bawait\b/.test(code);
+
+  if (mode === 'b' && !hasMaybeTLA) {
+    code = (await babel.transformAsync(code, { plugins: transformPlugins })).code;
   }
 
-  code = processTopLevelAwait(code) || code; // it returns null when no TLA is found
+  if (hasMaybeTLA) { // there's maybe a TLA await
+    const iiafe = processTopLevelAwait(code);
+    if (iiafe) { // there's a TLA
+      if (mode === 'b') {
+        code = (await babel.transformFromAstAsync(iiafe, code, { plugins: transformPlugins })).code;
+      } else {
+        code = babelGenerator(iiafe).code;
+      }
```

Comment:
Please add a tests for each of these:

```
user: n> var x = await Promise.resolve(true); x
jellobot: (okay) true
```

```
user: n> var x = await Promise.resolve(true); if (x) {}
jellobot: (okay) undefined
```

And the same exact code input / output but with `b>` instead of `n>`. I think that'll cover both of these branches and the concern in my other comment in this review.